modules/tensor_mechanics/src/materials/ComputeThermalExpansionEigenstrainBase.C

```diff
@@ -23,6 +23,9 @@ ComputeThermalExpansionEigenstrainBaseTempl<is_ad>::validParams()
   params.addParam<bool>("use_old_temperature",
                         false,
                         "Flag to optionally use the temperature value from the previous timestep.");
+  params.addParam<MaterialPropertyName>("mean_thermal_expansion_name",
+                                        "mean_thermal_expansion_coefficient",
```

Comment:
If nobody disagrees, I'd say go ahead and add `coefficient` to make it completely clear.

I'm not sure whether it's really worth the trouble of only computing this if requested. It's just a single scalar with no state, and the cost of computing it is pretty minimal.

Reply:
So I'm trying to optionally set a material property. In the constructor I've got
```
_calc_mean_CTE(parameters.isParamSetByUser("mean_thermal_expansion_coefficient_name")),
_mean_thermal_expansion_coefficient(_calc_mean_CTE ?  this->template declareProperty<Real>(
      this->template getParam<MaterialPropertyName>("mean_thermal_expansion_coefficient_name")) :
    nullptr)
```

This doesn't work as `nullptr` isn't compatible with MaterialProperty. What do I put in place of "nullptr"?

I was planning to use `_calc_mean_CTE` with if statements to skip assignments of values. Not sure if that will work either.

Reply:
Case and point is one of the failing bison tests on this PR. It is so easy to avoid this type of breakage by not creating that new property unless it is explicitly requested.

Reply:
That makes sense and I agree.

Do you have an example of this happening in another class that I could look at? I've never made an optional material property that doesn't exist when not asked for. I don't want `outputs=all` to throw an error from a material property that doesn't exist.

Reply:
I think I *might* have figured it out. 🤞